.gitignore

```diff
@@ -1,3 +1,45 @@
+# phpstorm project files
+.idea
+
+# netbeans project files
+nbproject
+
+# zend studio for eclipse project files
+.buildpath
+.project
+.settings
+
+# windows thumbnail cache
+Thumbs.db
+
+# composer vendor dir
 /vendor
+
+# composer itself is not needed
+composer.phar
+# composer lock is not needed in libraries https://getcomposer.org/doc/02-libraries.md#lock-file
+composer.lock
```

Comment:
The `composer.lock` file is included for assist with local development. While it's true imported projects ignore the lock file, it is not ignored when testing.